SessionMessagingKit/Database/Models/SessionThread.swift

```diff
@@ -293,60 +424,90 @@ public extension SessionThread {
         _ db: Database,
         type: SessionThread.DeletionType,
         threadId: String,
-        calledFromConfigHandling: Bool
+        calledFromConfigHandling: Bool,
+        using dependencies: Dependencies
     ) throws {
         try deleteOrLeave(
             db,
             type: type,
             threadIds: [threadId],
-            calledFromConfigHandling: calledFromConfigHandling
+            calledFromConfigHandling: calledFromConfigHandling,
+            using: dependencies
         )
     }
     
     static func deleteOrLeave(
         _ db: Database,
         type: SessionThread.DeletionType,
         threadIds: [String],
-        calledFromConfigHandling: Bool
+        calledFromConfigHandling: Bool,
+        using dependencies: Dependencies
     ) throws {
         let currentUserPublicKey: String = getUserHexEncodedPublicKey(db)
         let remainingThreadIds: Set<String> = threadIds.asSet().removing(currentUserPublicKey)
         
         switch type {
-            case .hideContactConversationAndDeleteContent:
+            case .hideContactConversation:
+                _ = try SessionThread
+                    .filter(ids: threadIds)
+                    .updateAllAndConfig(
+                        db,
+                        SessionThread.Columns.pinnedPriority.set(to: LibSession.hiddenPriority),
+                        SessionThread.Columns.shouldBeVisible.set(to: false),
+                        calledFromConfig: calledFromConfigHandling,
+                        using: dependencies
+                    )
+                
+            case .hideContactConversationAndDeleteContentDirectly:
                 // Clear any interactions for the deleted thread
                 _ = try Interaction
                     .filter(threadIds.contains(Interaction.Columns.threadId))
                     .deleteAll(db)
                 
+                // Hide the threads
+                _ = try SessionThread
+                    .filter(ids: threadIds)
+                    .updateAllAndConfig(
+                        db,
+                        SessionThread.Columns.pinnedPriority.set(to: LibSession.hiddenPriority),
+                        SessionThread.Columns.shouldBeVisible.set(to: false),
+                        calledFromConfig: calledFromConfigHandling,
+                        using: dependencies
+                    )
+            
+            case .deleteContactConversationAndMarkHidden:
+                _ = try SessionThread
+                    .filter(ids: remainingThreadIds)
+                    .deleteAll(db)
+                
                 // We need to custom handle the 'Note to Self' conversation (it should just be
                 // hidden locally rather than deleted)
                 if threadIds.contains(currentUserPublicKey) {
+                    // Clear any interactions for the deleted thread
+                    _ = try Interaction
```

Comment:
Ah, this will technically actually work as is (just since the interactions for all other threads would have been cascade deleted when deleting their threads so would just be ignored), but it's also unclear so I'll push up a tweak

Essentially in the case of `deleteContactConversationAndMarkHidden` we still don't want to _actually_ delete the "Note to Self" conversation so the deletion _just for the "Note to Self" conversation_ should behave the same as `hideContactConversationAndDeleteContentDirectly`

The filter should just be `.filter(Interaction.Columns.threadId == currentUserPublicKey)`